### PR TITLE
create workspace and mipidsi-async placeholder

### PR DESCRIPTION
**Pull request by almindor**
First step in making an async version of mipidsi. I think it'd be best to have it as a separate crate. If code-reuse that's ergonomic can be found we can perhaps move sections into a common library or such.

This should enable moving forward for #60 and #63/

This PR:

1. Moves `mipidsi` crate itself into a subfolder
2. Creates a workspace 
3. Adds a placeholder crate in `mipidsi-async` for that purpose
4. Adds a generic README pointing to individual crates

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,65 +1,5 @@
 # mipidsi
 
-<p align="center">
-    <a href="https://crates.io/crates/mipidsi"><img src="https://img.shields.io/crates/v/mipidsi.svg" alt="Crates.io"></a>
-    <a href="https://docs.rs/mipidsi"><img src="https://docs.rs/mipidsi/badge.svg" alt="Docs.rs"></a>
-    <a href="https://matrix.to/#/#rust-embedded-graphics:matrix.org"><img src="https://img.shields.io/matrix/rust-embedded-graphics:matrix.org" alt="Discuss driver on embedded-graphics on Matrix"></a>
-</p>
+Repository for the [mipidsi](mipidsi/README.md) and [mipidsi-sync](mipidsi-async/README.md) crates.
```
Comment (rfuest) [inline README.md:3]:
```suggestion
Repository for the [mipidsi](mipidsi/README.md) and [mipidsi-async](mipidsi-async/README.md) crates.
```

File: README.md [hunk 2]
```diff
@@ -1,65 +1,5 @@
 # mipidsi
```
Comment (rfuest) [inline README.md:1]: IMO the main README should cover both crates. Having an almost empty page when visiting the project on GitHub isn't a great experience for first time users.

The main README should at least contain an overview of what the purpose of the crates is, which is fundamentally the same for both. And also a description of the difference between the two crates to help the user choose the version they want to use. Details that only apply to one of the crates can then be added to the crate specific READMEs.,

File: mipidsi-async/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+[package]
+name = "mipidsi-async"
+description = "MIPI Display Serial Interface generic driver async version"
```
Comment (rfuest) [inline mipidsi-async/Cargo.toml:3]: We should change the description, because we don't implement MIPI DSI.

File: mipidsi-async/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+version = "0.1.0"
+authors = ["Ales Katona <almindor@gmail.com>"]
+edition = "2018"
+license = "MIT"
+repository = "https://github.com/almindor/mipidsi"
+keywords = ["embedded-hal-driver", "mipi", "dsi", "display"]
```
Comment (rfuest) [inline mipidsi-async/Cargo.toml:9]: Perhaps?
```suggestion
keywords = ["embedded-hal-driver", "mipi", "dcs", "display"]
```

File: mipidsi/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+[package]
```
Comment (rfuest) [inline mipidsi/Cargo.toml:1]: See comments for the `mipidsi-async` version of `Cargo.toml`.

File: mipidsi/README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+// create a DisplayInterface from SPI and DC pin, with no manual CS control
+let di = SPIInterfaceNoCS::new(spi, dc);
+// create the ILI9486 display driver in rgb666 color mode from the display interface and use a HW reset pin during init
+let mut display = Builder::ili9486_rgb666(di)
+    .init(&mut delay, Some(rst))?; // delay provider from your MCU
+// clear the display to black
+display.clear(Rgb666::BLACK)?;
+```
+
+License: MIT
```
Comment (rfuest) [inline mipidsi/README.md:60]: Maybe out of scope, but I hadn't noticed this before: Why is the license part of the "Troubleshooting" section?

I would suggest to change it to something similar to https://github.com/embedded-graphics/embedded-graphics#license (without the Apache license part). Adding a `LICENCE` file to the repo will also show the project license in the "About" section in the sidebar on GitHub.

File: mipidsi/README.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
+# mipidsi
```
Comment (rfuest) [inline mipidsi/README.md:1]: I assume this readme is unchanged? The GitHub interface didn't show a diff because the file has been moved.

**Review by rfuest — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (almindor) [reply README.md:1]: Thanks for all the suggestions. I've added the top level readme and fixed the licensing and DSI -> DCS issues.

Lmk if there's anything else that still seems off.

**Review by rfuest — APPROVED** [review]
LGTM

**Pull request merged**